src/lxc/attach.c

```diff
@@ -1512,9 +1512,18 @@ int lxc_attach(const char *name, const char *lxcpath,
 
 int lxc_attach_run_command(void* payload)
 {
+	int ret;
 	lxc_attach_command_t* cmd = (lxc_attach_command_t*)payload;
 
-	execvp(cmd->program, cmd->argv);
+	ret = execvp(cmd->program, cmd->argv);
+	if (ret < 0) {
+		switch (errno) {
+		case ENOEXEC:
+			return 126;
+		case ENOENT:
+			return 127;
+		}
+	}
 
 	SYSERROR("Failed to exec \"%s\"", cmd->program);
```

Comment:
Shouldn't SYSERROR be reported *before* the return case above? Otherwise nothing gets logged any more for ENOEXEC and ENOENT!

Reply:
Yeah, very obvious oversight on my part. Thanks!

Reply:
fixed